rio_tiler/reader.py

```diff
@@ -214,32 +211,31 @@ def read(
         # We only add dataset statistics if we have them for all the indexes
         dataset_statistics = stats if len(stats) == len(indexes) else None
 
+        # TODO: DEPRECATED, masked array are already using bool
         if force_binary_mask:
-            mask = numpy.where(mask != 0, numpy.uint8(255), numpy.uint8(0))
+            pass
+            # mask = numpy.where(mask != 0, numpy.uint8(255), numpy.uint8(0))
```

Comment:
This is the biggest breaking change! now that we use rasterio `masked` and numpy.ma.MaskedArray for the PER_BAND mask representation we won't support non-binary mask. It will always be True (0) or False (255). 

This is a major downside for people using non-binary alpha bands, but sometimes we have to take hard decisions. (ref: https://github.com/rasterio/rasterio/issues/1721#issuecomment-586547617)